src/ValidationExemption.php

```diff
@@ -0,0 +1,247 @@
+			$node instanceof Element
+			&&
+			$node->hasAttribute( self::PX_VERIFIED_TAG_ATTRIBUTE )
+		) {
+			return true;
+		}
+
+		// Check attribute.
+		if (
+			$node instanceof DOMAttr
+			&&
+			self::check_for_attribute_token_list_membership( $node, self::PX_VERIFIED_ATTRS_ATTRIBUTE )
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Mark node as being PX-verified.
+	 *
+	 * @param DOMNode $node Node.
+	 * @return bool Whether successful.
+	 */
+	public static function mark_node_as_px_verified( DOMNode $node ) {
+
+		if ( $node instanceof Element ) {
+			if ( ! $node->hasAttribute( self::PX_VERIFIED_TAG_ATTRIBUTE ) ) {
+				if ( ! $node->ownerDocument ) {
+					return false; // @codeCoverageIgnore
+				}
+				$node->setAttributeNode( $node->ownerDocument->createAttribute( self::PX_VERIFIED_TAG_ATTRIBUTE ) );
+			}
+			return true;
+		}
+
+		if ( $node instanceof DOMAttr ) {
+			$element = $node->parentNode;
+			if ( ! $element instanceof Element ) {
+				return false; // @codeCoverageIgnore
+			}
+
+			$attr_value = $element->getAttribute( self::PX_VERIFIED_ATTRS_ATTRIBUTE );
+			if ( $attr_value ) {
+				$attr_value .= ' ' . $node->nodeName;
+			} else {
+				$attr_value = $node->nodeName;
+			}
+
+			$element->setAttribute( self::PX_VERIFIED_ATTRS_ATTRIBUTE, $attr_value );
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Mark node as not being AMP-validated.
+	 *
+	 * @param DOMNode $node Node.
+	 * @return bool Whether successful.
+	 */
+	public static function mark_node_as_amp_unvalidated( DOMNode $node ) {
+
+		if ( $node instanceof Element ) {
+			if ( ! $node->hasAttribute( self::AMP_UNVALIDATED_TAG_ATTRIBUTE ) ) {
+				if ( ! $node->ownerDocument ) {
+					return false; // @codeCoverageIgnore
+				}
+				$node->setAttributeNode( $node->ownerDocument->createAttribute( self::AMP_UNVALIDATED_TAG_ATTRIBUTE ) );
+			}
+			return true;
+		}
+
+		if ( $node instanceof DOMAttr ) {
+			$element = $node->parentNode;
+			if ( ! $element instanceof Element ) {
+				return false; // @codeCoverageIgnore
+			}
+
+			$attr_value = $element->getAttribute( self::AMP_UNVALIDATED_ATTRS_ATTRIBUTE );
+			if ( $attr_value ) {
+				$attr_value .= ' ' . $node->nodeName;
+			} else {
+				$attr_value = $node->nodeName;
+			}
+
+			$element->setAttribute( self::AMP_UNVALIDATED_ATTRS_ATTRIBUTE, $attr_value );
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Check whether AMP is unvalidated for node.
+	 *
+	 * This means that it is exempted from AMP validation and it may negatively impact PX.
+	 *
+	 * @param DOMNode $node Node.
+	 * @return bool Whether node marked as being AMP-unvalidated.
+	 */
+	public static function is_amp_unvalidated_for_node( DOMNode $node ) {
+
+		// Prevent removing a tag which was exempted from validation.
+		if (
+			$node instanceof \DOMElement
+			&&
+			$node->hasAttribute( self::AMP_UNVALIDATED_TAG_ATTRIBUTE )
+		) {
+			return true;
+		}
+
+		// Check attribute.
+		if (
+			$node instanceof DOMAttr
```

Comment:
If the check for the object type is separate, the second can in some instance be skipped, as in the other examples via an `elseif` instead of an `if`.

Reply:
Like 96a4c113a2c746b3a74cab6b9c9aaf1ea162879e?